pywatts/wrapper/statsmodels_wrapper.py

```diff
@@ -0,0 +1,164 @@
+
+    :param module: The statsmodels module to wrap
+    :type module: sm.tsa.base.tsa_model.TimeSeriesModel
+    :param name: The name of the module
+    :type name: str
+    :param module_kwargs: The module keyword arguments necessary for creating the statsmodel module
+    :type module_kwargs: dict
+    :param fit_kwargs: The optional fit keyword arguments for fitting the model
+    :type fit_kwargs: dict
+    :param predict_kwargs: The optional predict keyword arguments for predicting with the model (except start and end)
+    :type predict_kwargs: dict
+    :param model_params: The fitted model parameters
+    :type xr.Dataset
+    """
+
+    def __init__(self, module: sm.tsa.base.tsa_model.TimeSeriesModel, name: str = None, module_kwargs=None,
+                 fit_kwargs=None, predict_kwargs=None, model_params=None):
+        if name is None:
+            name = module.__class__.__name__
+        super().__init__(name)
+        if module_kwargs is None:
+            module_kwargs = {}
+        if fit_kwargs is None:
+            fit_kwargs = {}
+        if predict_kwargs is None:
+            predict_kwargs = {}
+        self.module = module
+        self.module_kwargs = module_kwargs
+        self.fit_kwargs = fit_kwargs
+        self.predict_kwargs = predict_kwargs
+        self.model_params = model_params
+
+    def get_params(self) -> Dict[str, object]:
+        """
+        Returns the parameters of the statsmodels module.
+
+        :return: A dict containing the module keyword arguments, the fit keyword arguments, the predict keyword
+        arguments and the fitted model parameters
+        :rtype: dict
+        """
+        return {
+            "module_kwargs": self.module_kwargs,
+            "fit_kwargs": self.fit_kwargs,
+            "predict_kwargs": self.predict_kwargs,
+            "model_params": self.model_params
+        }
+
+    def set_params(self, module_kwargs=None, fit_kwargs=None, predict_kwargs=None, model_params=None):
+        """
+        Set the parameters of the statsmodels wrapper
+
+        :param module_kwargs: keyword arguments for the statsmodel module.
+        :param fit_kwargs: keyword arguments for the fit method.
+        :param predict_kwargs: keyword arguments for the predict method.
+        :param model_params: parameters of the fitted model
+        """
+        if module_kwargs:
+            self.module_kwargs = module_kwargs
+        if fit_kwargs:
+            self.fit_kwargs = fit_kwargs
+        if predict_kwargs:
+            self.predict_kwargs = predict_kwargs
+        if model_params:
+            self.model_params = model_params
+
+    def fit(self, x: xr.Dataset, y: xr.Dataset = None):
+        """
+        Fits the statsmodels module
+
+        :param x: input data
+        :param y: target data (not needed)
+        """
+        x = self._dataset_to_statsmodels_input(x)
+        self.model_params = self.module(x, **self.module_kwargs).fit(**self.fit_kwargs).params
```

Comment:
Why do you initialise or call on the statsmodels module: `self.module(x, **self.module_kwargs)`?
I would assume to pass an initialised model to statsmodels so simply use `self.module.fir(**self.fit_kwargs)` in the wrapper class?

But I have to say that I never used statsmodels before... So maybe this is needed before calling the `fit` method. However, I do not found any documentation on the `__call__` method in the docs [[Statsmodels TimeSeriesModel](https://tedboy.github.io/statsmodels_doc/generated/generated/statsmodels.tsa.base.tsa_model.TimeSeriesModel.html)].

Reply:
For initialising the module you need the dataset in statsmodel... E.g., see [OLS in statsmodel](https://www.statsmodels.org/devel/examples/notebooks/generated/ols.html)